src/java.base/share/classes/jdk/internal/util/DecimalDigits.java

```diff
@@ -443,4 +446,45 @@ private static void uncheckedPutCharUTF16(byte[] buf, int charPos, int c) {
         assert charPos >= 0 && charPos < (buf.length >> 1);
         UNSAFE.putCharUnaligned(buf, ARRAY_BYTE_BASE_OFFSET + ((long) charPos << 1), (char) c);
     }
+
+    /**
+     * Appends the two-digit string representation of the {@code int}
+     * argument to the given {@code StringBuilder}.
+     * <p>
+     * The integer {@code v} is formatted as two decimal digits.
+     * Values from 0 to 9 are formatted with a leading zero (e.g., 5 becomes "05"),
+     * and values from 10 to 99 are formatted as regular two-digit numbers.
+     * If the value is outside the range 0-99, the behavior is unspecified.
+     *
+     * @param buf the {@code StringBuilder} to append to.
+     * @param v the {@code int} value (should be between 0 and 99 inclusive).
+     */
+    public static void appendPair(StringBuilder buf, int v) {
+        int packed = DIGITS[v & 0x7f];
+        buf.append(
+                JLA.uncheckedNewStringWithLatin1Bytes(
+                        new byte[] {(byte) (packed & 0xFF), (byte) (packed >> 8)}));
+    }
+
+    /**
+     * Appends the four-digit string representation of the {@code int}
+     * argument to the given {@code StringBuilder}.
+     * <p>
+     * The integer {@code v} is formatted as four decimal digits.
+     * Values from 0 to 9 are formatted with leading zeros (e.g., 5 becomes "0005"),
+     * values from 10 to 99 become three leading zeros (e.g., 25 becomes "0025"),
+     * values from 100 to 999 become two leading zeros (e.g., 123 becomes "0123"),
+     * and values from 1000 to 9999 become one leading zero or no leading zeros.
+     * If the value is outside the range 0-9999, the behavior is unspecified.
+     *
+     * @param buf the {@code StringBuilder} to append to.
+     * @param v the {@code int} value (should be between 0 and 9999 inclusive).
+     */
+    public static void appendQuad(StringBuilder buf, int v) {
+        int y01 = v / 100;
```

Comment:
Same assert bound check request

Reply:
To maintain consistency with other putPair methods in DecimalDigits, trust the caller and do not perform assert checks, as assert checks can increase codeSize and potentially impact performance.